### PR TITLE
Add protocol as a configuration option for the dd-trace-agent URL

### PR DESCRIPTION
**Pull request by RyanGordon**
This is useful for scenarios where a SSL-terminating load balancer sits in front of the datadog-agent's trace endpoint

This resolves issue #415

### CODE REVIEW

**Comment by RyanGordon** [comment]
Hi @rochdev sure will make the change.

What should the precedence order be when the `port` or `hostname` config option is set along side the `DD_TRACE_AGENT_URL` env variable?

File: test/config.spec.js [hunk 1]
```diff
@@ -114,7 +114,7 @@ describe('Config', () => {
       enabled: false,
       debug: true,
       hostname: 'agent',
-      overrideUrl: 'https://agent2:443',
+      overrideUrl: 'https://agent2:7777',
```
Comment (rochdev) [inline test/config.spec.js]: Why not just `url` or `endpoint`?

Comment (RyanGordon) [reply test/config.spec.js]: Yeah, i thought it might be a little confusing with just `url`, but the update docs should clear that up incase its not clear

**Comment by RyanGordon** [comment]
Great, that makes sense. This should be all fixed up now

**Pull request merged**